Microsoft.Toolkit.Uwp.UI.Controls/BladeView/BladeView.Properties.cs

```diff
@@ -27,12 +29,66 @@ public partial class BladeView
         public static readonly DependencyProperty ActiveBladesProperty = DependencyProperty.Register(nameof(ActiveBlades), typeof(IList<BladeItem>), typeof(BladeView), new PropertyMetadata(null));
 
         /// <summary>
+        /// Identifies the <see cref="FullScreenBlades"/> attached property.
+        /// </summary>
+        public static readonly DependencyProperty FullScreenBladesProperty = DependencyProperty.RegisterAttached(nameof(FullScreenBlades), typeof(bool), typeof(BladeView), new PropertyMetadata(false, FullScreenBladesChangedCallback));
+
+        /// <summary>
         /// Gets or sets a collection of visible blades
         /// </summary>
         public IList<BladeItem> ActiveBlades
         {
             get { return (IList<BladeItem>)GetValue(ActiveBladesProperty); }
             set { SetValue(ActiveBladesProperty, value); }
         }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether blades are full screen
+        /// </summary>
+        public bool FullScreenBlades
+        {
+            get { return (bool)GetValue(FullScreenBladesProperty); }
+            set { SetValue(FullScreenBladesProperty, value); }
+        }
+
+        private static void FullScreenBladesChangedCallback(DependencyObject dependencyObject, DependencyPropertyChangedEventArgs e)
+        {
+            // Check if value changed
+            if (e.NewValue != e.OldValue)
+            {
+                var bladeView = (BladeView)dependencyObject;
+                var bladeScrollViewer = bladeView.GetScrollViewer();
+
+                if (bladeView.FullScreenBlades)
+                {
+                    // Cache previous values of blade items properties (width & height)
+                    bladeView._cachedBladeItemSizes.Clear();
+                    foreach (BladeItem bladeItem in bladeView.Items)
+                    {
+                        bladeView._cachedBladeItemSizes.Add(bladeItem, new Vector2((float)bladeItem.Width, (float)bladeItem.Height));
+                    }
+
+                    // Change ScrollView behavior
+                    bladeScrollViewer.HorizontalSnapPointsType = SnapPointsType.MandatorySingle;
```

Comment:
Change to a visual state by using the VisualStateManager: `VisualStateManager.GoToState(this, "FullScreen", false);`
You create the visual states in the xaml definition of the control
Example:
```
<VisualStateManager.VisualStateGroups>
    <VisualStateGroup x:Name="BladeModeStates">
        <VisualState x:Name="Normal"/> <!-- No changes required. When in this state propertieswill revert to their original values -->
        <VisualState x:Name="FullScreen">
            <Storyboard>
                <ObjectAnimationUsingKeyFrames Storyboard.TargetProperty="HorizontalSnapPointsType" Storyboard.TargetName="ScrollViewer">
                    <DiscreteObjectKeyFrame KeyTime="0" Value="MandatorySingle"/>
                </ObjectAnimationUsingKeyFrames>
            </Storyboard>
        </VisualState>
    </VisualStateGroup>
</VisualStateManager.VisualStateGroups>
```